src/analysis/processing/qgsalgorithmmultiringconstantbuffer.cpp

```diff
@@ -0,0 +1,147 @@
+}
+
+void QgsMultiRingConstantBufferAlgorithm::initParameters( const QVariantMap & )
+{
+  std::unique_ptr< QgsProcessingParameterNumber> rings = qgis::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "RINGS" ),
+      QObject::tr( "Number of rings" ), QgsProcessingParameterNumber::Integer,
+      1, false, 0 );
+  rings->setIsDynamic( true );
+  rings->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "RINGS" ), QObject::tr( "Number of rings" ), QgsPropertyDefinition::IntegerPositive ) );
+  rings->setDynamicLayerParameterName( QStringLiteral( "INPUT" ) );
+  addParameter( rings.release() );
+
+  std::unique_ptr< QgsProcessingParameterNumber> distance = qgis::make_unique< QgsProcessingParameterNumber >( QStringLiteral( "DISTANCE" ),
+      QObject::tr( "Distance between rings" ), QgsProcessingParameterNumber::Double,
+      1, false, 0 );
+  distance->setIsDynamic( true );
+  distance->setDynamicPropertyDefinition( QgsPropertyDefinition( QStringLiteral( "DISTANCE" ), QObject::tr( "Distance between rings" ), QgsPropertyDefinition::DoublePositive ) );
+  distance->setDynamicLayerParameterName( QStringLiteral( "INPUT" ) );
+  addParameter( distance.release() );
+}
+
+bool QgsMultiRingConstantBufferAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  mRingsNumber = parameterAsInt( parameters, QStringLiteral( "RINGS" ), context );
+  mDynamicRingsNumber = QgsProcessingParameters::isDynamic( parameters, QStringLiteral( "RINGS" ) );
+  if ( mDynamicRingsNumber )
+    mRingsNumberProperty = parameters.value( QStringLiteral( "RINGS" ) ).value< QgsProperty >();
+
+  mDistance = parameterAsDouble( parameters, QStringLiteral( "DISTANCE" ), context );
+  mDynamicDistance = QgsProcessingParameters::isDynamic( parameters, QStringLiteral( "DISTANCE" ) );
+  if ( mDynamicDistance )
+    mDistanceProperty = parameters.value( QStringLiteral( "DISTANCE" ) ).value< QgsProperty >();
+
+  return true;
+}
+
+QgsFields QgsMultiRingConstantBufferAlgorithm::outputFields( const QgsFields &inputFields ) const
+{
+  QgsFields fields = inputFields;
+  fields.append( QgsField( QStringLiteral( "ringId" ), QVariant::Int, QString(), 10, 0 ) );
+  fields.append( QgsField( QStringLiteral( "distance" ), QVariant::Double, QString(), 20, 6 ) );
+  return fields;
+}
+
+QgsFeatureList QgsMultiRingConstantBufferAlgorithm::processFeature( const QgsFeature &feature, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  double currentDistance = 0;
+  QgsGeometry outputGeometry, previousGeometry;
+
+  int rings = mRingsNumber;
+  if ( mDynamicRingsNumber )
+    rings = mRingsNumberProperty.valueAsInt( context.expressionContext(), rings );
+
+  double distance = mDistance;
+  if ( mDynamicDistance )
+    distance = mDistanceProperty.valueAsDouble( context.expressionContext(), distance );
+
+  QgsFeatureList outputs;
+
+  for ( int i = 1; i <= mRingsNumber; ++i )
```

Comment:
shouldn't this be rings, not mRingsNumber?

Reply:
Oops... fixed